src/main/java/org/springframework/data/neo4j/core/ReactiveNeo4jClient.java

```diff
@@ -273,6 +257,61 @@ public ReactiveNeo4jClient build() {
 		Mono<ResultSummary> run();
 	}
 
+	/**
+	 * Contract for a runnable query specification which still can be bound to a specific database and an impersonated user.
+	 *
+	 * @since 6.2
+	 */
+	interface UnboundRunnableSpec extends RunnableSpec {
+
+		/**
+		 * Pins the previously defined query to a specific database. A value of {@literal null} chooses the default
+		 * database. The empty string {@literal ""} is not permitted.
```

Comment:
Minor thing but worth a thought: Interesting constellation. We say that an empty string is not supported but this is only checked in the implementation.
Applies also to the user and imperative branch.